src/core/provider.cc

```diff
@@ -176,39 +181,46 @@ InferRequestProvider::GetNextInputContent(
       return Status(
           RequestStatusCode::INTERNAL, "unexpected input '" + name + "'");
     }
-
```

Comment:
In the provider the raw buffer if fetched from the input_buffer content. We'll need a check here to read from the shared memory buffer instead of from the input_buffer here. Similarly for the raw output.

Reply:
Fixed shared memory input for grpc. Fixing for http as well.